hw/ip/pattgen/doc/_index.md

```diff
@@ -0,0 +1,137 @@
+---
+title: "PATTGEN HWIP Technical Specification"
+---
+
+# Overview
+
+This document specifies PATTGEN hardware IP (HWIP) functionality.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification/index.md" >}})
+See that document for integration overview within the broader top-level system.
+
+## Features
+
+- Generates time-dependent patterns on two (2) channels, each with its own clock.
+   - In each channel, data is transmitted serially on a one-bit data (`pda`) output, which is synchronous to a corresponding parallel clock signal (`pcl`).
+   - The channels may operate independently or synchronously with each other.
+- Each output channel supports the following configuration registers:
+    - Pattern data per output (up to 64 bits of data).
+    - Pattern length per output (between 1 and 64 bit).
+    - 32-bit pre-divider to derive pattern clock from I/O clock (minimum ratio: 2).
+    - A 10-bit repeat counter
```

Comment:
I'd go for a more feature-like description here, e.g. "Each pattern can be repeated up to 2^10-1 times."

Reply:
Sure.  Thanks for the suggestion.

Reply:
Done.